### DISCUSSION TITLE
Find unit tests in location other than `integrationFolder`

### DISCUSSION THREAD

**Discussion by michaelhays** [cypress-io/cypress#16307] [Questions and Help] [answered]
Hey there, I'd like Cypress to find unit tests in a location other than `integrationFolder` (`cypress/integration` by default).

Here's my desired project structure:

```
myproject/
├── cypress/
│   └── integration/
│       └── e2e.spec.js
└── src/
    └── SomeComponent/
        ├── index.vue
        ├── utils.js
        └── utils.spec.js
```

(`utils.spec.js` has unit tests for `utils.js`)

For me, this is far preferable to putting the unit tests in `cypress/integration`, since we can keep them directly next to the source code that they test.

@brian-mann actually affirms this sentiment in [this comment](https://github.com/cypress-io/cypress/issues/1189#issuecomment-357990472) (which I agree with):

> The idea is that e2e tests are not strictly coupled to source code and therefore belong in their own folder instead of next to the code (like unit tests are)

However, I haven't found any way to actually configure Cypress to find my unit tests in `src/` rather than `cypress/integration/`. The [configuration](https://docs.cypress.io/guides/references/configuration#Folders-Files) on this does not offer anything (no mention of "unit" in that entire page), and the [Writing and Organizing Tests page](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests) claims in the intro that the document explains "How Cypress handles unit tests vs integration tests", but a search for "unit" in that page doesn't actually reveal anything.

Is this currently possible in Cypress, or should I open a feature request for it?

**Comment by bahmutov** [comment, marked as answer]
Have you tried using "component testing" approach https://on.cypress.io/component-testing - it can be used for running unit tests too.  You can control the folder and point at the specs next to the source files

```json
{
  "testFiles": "cypress/integration/*.spec.js",
  "component": {
    "componentFolder": "src",
    "testFiles": ".*/__tests__/.*spec.tsx",
    "viewportHeight": 500,
    "viewportWidth": 700
  }
}
```

**Reply by michaelhays** [reply]
I haven't, I'll give it a try and report back. Thanks!

**Reply by michaelhays** [reply]
Yep works great, appreciate the help!
